app/common/expressions/managed.py

```diff
@@ -496,6 +496,57 @@ def build_from_form(form: "_ManagedExpressionForm", question: "Question") -> "Is
         return IsNo(question_id=question.id)
 
 
+@register_managed_expression
+class HasDomain(ManagedExpression):
+    name: ClassVar[ManagedExpressionsEnum] = ManagedExpressionsEnum.HAS_DOMAIN
+    supported_condition_data_types: ClassVar[set[QuestionDataType]] = {QuestionDataType.EMAIL}
+    supported_validator_data_types: ClassVar[set[QuestionDataType]] = {QuestionDataType.EMAIL}
+
+    _key: ManagedExpressionsEnum = name
+
+    question_id: UUID
+
+    domain: str
+
+    @property
+    def description(self) -> str:
+        return "has the domain"
+
+    @property
+    def message(self) -> str:
+        return f"The answer must be from “{self.domain}”"
+
+    @property
+    def statement(self) -> str:
+        # fixme: this could register a function with simpleeval to explicitly check the domain part of
+        #       a given email input
+        return f"'{self.domain}' in {self.safe_qid}"
```

Comment:
Yeah this sounds clever - would be interested in seeing an example of it. I've felt the lack on another expression I was writing; feels like you've got a thought and it'd be cool to see that explored/proved 🙇